ndd_slurm.py

```diff
@@ -178,26 +187,30 @@ def get_ssh_slave_ndd_cmd(args, slave, slaves):
     return cmd
 
 
-def get_ssh_slave_cmds(args):
+def get_ssh_cmds(args):
     SSH = ['ssh', '-tt', '-o', 'PasswordAuthentication=no']
     slaves = args.d
-    cmds = [(SSH + [slave] + get_ssh_slave_ndd_cmd(args, slave, slaves))\
+    cmds = [(SSH + [slave] + get_ssh_slave_cmd(args, slave, slaves))\
             for slave in slaves]
     return cmds
 
 
 def run_ssh_master(args):
-    cmds = [get_master_ndd_cmd(args)] + get_ssh_slave_cmds(args)
+    cmds = [get_master_ndd_cmd(args)] + get_ssh_cmds(args)
     procs = {proc.pid: proc for proc in map(init_process, cmds)}
     return wait(procs)
 
 
 def run_ssh_slave(args, env):
-    return subprocess.call(get_ssh_slave_ndd_cmd(args, env))
+    slave = args.c
+    slaves = args.S.split(',')
+    return subprocess.call(get_ssh_slave_ndd_cmd(args, slave, slaves))
 
 
 def main(raw_args, env):
-    if len(raw_args) > 0 and raw_args[0] == '-S':
+    if len(raw_args) > 0 and raw_args[0] == '-S' and '-H' in raw_args:
```

Comment:
This can be combined:
```
if len(raw_args > 0) and raw_args[0] == '-S':
    if '-H' in raw_args:
        return run_ssh_slave(...)
    else:
        return run_slave(...)
```